src/System.Threading.Tasks/tests/Task/TaskCancelWaitTests.cs

```diff
@@ -2748,6 +2748,38 @@ public static void TaskCancelWait141()
             test.RealRun();
         }
 
+        [Fact]
+        [OuterLoop]
+        public static void TaskCancelWait142()
+        {
+            // Verify that Task.Wait can return before all continuations scheduled
+            // with ExecuteSynchronously complete
+
+            Task task1 = new Task(() =>
+            {
+                Task.Delay(1).Wait();
+            });
+
+            CancellationTokenSource cts = new CancellationTokenSource();
+            Task task2 = task1.ContinueWith((_task) =>
+            {
+                while (!cts.IsCancellationRequested)
+                {
+                    Task.Delay(1).Wait();
+                }
+            }, TaskContinuationOptions.ExecuteSynchronously);
+
+            task1.Start();
+
+            // Wait should return once the task is complete, regardless of what other 
+            // continuations were scheduled off of it.
+            task1.Wait();
+            Task.WaitAll(task1);
```

Comment:
This WaitAll isn't necessary, as it won't really be testing anything... once the task has completed, a fast path in WaitAll will avoid doing any further waiting.  It'd be more interesting if the WaitAll were used instead of the Wait, e.g. if this were a [Theory] with a wait mode argument that would switch between these two.